public/app/features/plugins/admin/helpers.ts

```diff
@@ -345,3 +351,26 @@ function isDisabledSecretsPlugin(type?: PluginType): boolean {
 export function isLocalCorePlugin(local?: LocalPlugin): boolean {
   return Boolean(local?.signature === 'internal');
 }
+
+function getId(inputString: string): string {
+  const parts = inputString.split(' - ');
+  return parts[0];
+}
+
+function getPluginDetailsForFuzzySearch(plugins: CatalogPlugin[]): { [key: string]: string } {
+  return plugins.reduce((result: { [key: string]: string }, { id, name, type, orgName, info }: CatalogPlugin) => {
+    const keywordsForSearch = info.keywords?.join(' ').toLowerCase();
+    result[id] = `${id} - ${name} - ${type} - ${orgName} - ${keywordsForSearch}`;
+    return result;
+  }, {});
+}
+export function filterByKeyword(plugins: CatalogPlugin[], query: string) {
+  const dataArray = Object.values(getPluginDetailsForFuzzySearch(plugins));
+  let opts = {};
+  let uf = new uFuzzy(opts);
```

Comment:
you can pass `{}` directly to the constructor and save declaring the `opts` variable since we don't use it for anything else.